File: src/app/services/ISpeedSensorService.ts
import { PowerMeterServiceEvent } from "../model/PowerMeterServiceEvent";
import { SpeedSensorServiceEvent } from "../model/SpeedSensorServiceEvent";

export interface ISpeedSensorService {
    connect(): Promise<void>;
    disconnect(): Promise<void>;
    reconnectToLastConnected(): Promise<void>;
    selectNewDevice(): Promise<void>;
    subscribeForSpeedData(callback: (event: SpeedSensorServiceEvent) => void): void;
    unsubscribeForSpeedData(callback: (event: SpeedSensorServiceEvent) => void): void;
    readonly deviceSelected: boolean;
    readonly deviceName: string;
    readonly isConnected: boolean;
}

File: android/app/src/main/java/com/openbikecomputer/service/ISpeedSensorService.java
package com.openbikecomputer.service;

/**
 * Bluetooth cycling speed and cadence sensor. Ported from {@code ISpeedSensorService.ts}.
 */
public interface ISpeedSensorService {
    void selectNewDevice();

    void reconnectToLastConnected();

    void disconnect();

    boolean isConnected();

    boolean isDeviceSelected();

    String getDeviceName();

    void subscribeForSpeedData(OnSpeedData listener);

    void unsubscribeForSpeedData(OnSpeedData listener);
}
